frontend/src/static/js/api.js

```diff
@@ -138,26 +138,27 @@ window.getGroupName = function getGroupName(group, filterGroupSelection) {
 };
 
 window.api = {
-  loadJSON(fname) {
-    if (REPORT_ZIP) {
-      const zipObject = REPORT_ZIP.file(fname.slice(2));
+  async loadJSON(fname) {
+    if (window.REPORT_ZIP) {
+      const zipObject = window.REPORT_ZIP.file(fname.slice(2));
       if (zipObject) {
-        return zipObject.async('text').then((txt) => JSON.parse(txt));
+        try {
+          return JSON.parse(await zipObject.async('text'));
+        } catch (e) {
+          throw new Error('Json is invalid.');
+        }
+      } else {
+        throw new Error('Zip file is invalid.');
       }
-      return Promise.reject(new Error('Zip file is invalid.'));
     }
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.open('GET', fname);
-      xhr.onload = function xhrOnload() {
-        if (xhr.status === 200) {
-          resolve(JSON.parse(xhr.responseText));
-        } else {
-          reject(new Error('Unable to get file.'));
-        }
-      };
-      xhr.send(null);
-    });
+    try {
+      const response = await fetch(fname);
+      // Not directly returned in case response is not actually json.
+      const json = await response.json();
+      return json;
+    } catch (e) {
+      throw new Error('Json file not found.');
```

Comment:
Codewise OK to me. However, this one should not be an error as uploading a ZIP file is also in our normal use cases.

Reply:
Oh the error is just to tell the functions calling it that it didn't find any json to load.
Returning an empty string would cause some things to crash.

For example, it would probably be fine for `this.loadJSON(`${REPORT_DIR}/summary.json`)` to throw an error if `${REPORT_DIR}/summary.json` doesn't exist.

Using a different implementation would require further changes.

Reply:
The issue is that, this error will cause an alert dialog to be opened saying (json file not found), which should not appear in our design.

Reply:
Should be resolved